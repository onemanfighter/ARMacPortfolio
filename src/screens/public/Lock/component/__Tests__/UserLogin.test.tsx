import { fireEvent, render, screen } from '@testing-library/react';
import UserLoginComponent from '../UserLogin';
import { renderHook } from '@testing-library/react-hooks';
import { loginSelector, processStore } from '@processStore';

describe('UserLogin', () => {
  it('should return the default user login component', () => {
    const { container } = render(<UserLoginComponent />);

    expect(container).toMatchSnapshot();
  });

  it('should show spinner on click on login button', () => {
    const { container } = render(<UserLoginComponent />);

    expect(screen.queryByLabelText('spinner')).toBeNull();
    fireEvent.click(screen.getByLabelText('login-button'));

    expect(container).toMatchSnapshot();
    expect(screen.getByLabelText('spinner')).toBeDefined();
  });

  it('should show spinner on click on login button', async () => {
    const { container } = render(<UserLoginComponent />);

    fireEvent.click(screen.getByLabelText('user-button'));

    jest.advanceTimersByTimeAsync(1000);

    expect(
      screen.queryByText('Your password is required to enable Touch ID')?.style
        .opacity,
    ).toEqual('');
    expect(container).toMatchSnapshot();
  });

  it('should be able to enter password on password field', () => {
    const { container } = render(<UserLoginComponent />);

    fireEvent.change(screen.getByLabelText('password-text-input'), {
      target: { value: 'password-test-text' },
    });

    expect(container).toMatchSnapshot();
  });

  it('should be able to enter password on password field', async () => {
    const { result } = renderHook(() => processStore(loginSelector));
    const { container } = render(<UserLoginComponent />);

    expect(result.current.isLoggedIn).toBeFalsy();
    fireEvent.change(screen.getByLabelText('password-text-input'), {
      target: { value: 'Amit' },
    });
    fireEvent.click(screen.getByLabelText('login-button'));

    await jest.advanceTimersByTimeAsync(3000);

    expect(container).toMatchSnapshot();
    expect(result.current.isLoggedIn).toBeTruthy();
  });
});
