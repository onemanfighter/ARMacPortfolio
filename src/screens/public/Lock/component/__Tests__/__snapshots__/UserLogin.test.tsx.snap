// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogin should return the default user login component 1`] = `
<div>
  <div
    aria-label="user-login-component"
    class="css-1dvrt7t"
  >
    <button
      class="chakra-button css-y5bm5g"
      type="button"
    >
      <img
        alt="Profile"
        class="chakra-image css-13wtmhl"
        loading="lazy"
        src="https://avatars.githubusercontent.com/u/104697219?s=400&u=7ecc539c268755cfe0409ca5863773bb726387ba&v=4"
      />
      <p
        class="chakra-text css-1590av0"
      >
        Amit Raikwar
      </p>
    </button>
    <div
      class="chakra-input__group css-139cpa4"
      data-group="true"
    >
      <input
        class="chakra-input css-1362yyb"
        maxlength="20"
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <div
        class="chakra-input__right-element css-1ttnbqn"
      >
        <button
          aria-label="Login button"
          class="chakra-button css-15p997s"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <p
      class="chakra-text css-1ahyq4l"
    >
      Touch ID or Enter password
    </p>
  </div>
</div>
`;
